### PR TITLE
:ES6: anagram helpers

### PR DESCRIPTION
**Pull request by regiskuckaertz**
Some help for the crosswords enthusiasts.

I'll wait for @SiAdcock to merge his work on SVGs in tests, because I think they currently break them in this PR, and life is too short.

Who's responsible for client-side crosswords?

### CODE REVIEW

**Comment by sndrs** [comment]
better cc @MatthewJWalls on this

**Comment by desbo** [comment]
i built the anagram helper originally so i'll take a look

**Comment by regiskuckaertz** [comment]
@desbo thank you 😄  fyi, I replaced lodash helpers with native calls, since arrow functions close on `this` from the surrounding context.

**Comment by regiskuckaertz** [comment]
@desbo I can replace `React.createClass` with a subclass of `React.Component` if that makes sense

File: static/src/javascripts/projects/common/modules/crosswords/anagram-helper/clue-preview.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
+// @flow
+
+/* code smell: */
+/* eslint "react/no-array-index-key": "off" */
+
+import React from 'react/addons';
+
+type WeirdKey = ',' | '-';
```
Comment (desbo) [inline static/src/javascripts/projects/common/modules/crosswords/anagram-helper/clue-preview.js]: 😆

File: static/src/javascripts/projects/common/modules/crosswords/anagram-helper/main.js [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+        const clue = helpers.getAnagramClueData(
+            this.props.entries,
+            this.props.focussedEntry
+        );
+        const cells = helpers.cellsForClue(
+            this.props.entries,
+            this.props.focussedEntry
+        );
+        const entries = cells.map(
+            coords => this.props.grid[coords.x][coords.y]
```
Comment (desbo) [inline static/src/javascripts/projects/common/modules/crosswords/anagram-helper/main.js]: If you include the `x` and `y` coords here, you should be able to fix the code smell in the other two files by using the coordinates for the key.

**Comment by sndrs** [comment]
did you see #16842?

**Comment by regiskuckaertz** [comment]
lol, nope. time to close this one 😆

**Comment by MatthewJWalls** [comment]
I think this one is a more complete solution than my pr, so think we should just compare notes in the morning and see if we can merge them or something. How did we manage to step on each others toes with the es6  assignment system though?

**Comment by regiskuckaertz** [comment]
@MatthewJWalls hi - it's my bad. I was done with my list of modules and picked some at random—sadly the wrong ones 😭  are you around?

**Comment by sndrs** [comment]
can we merge this?

**Comment by regiskuckaertz** [comment]
Finally got around to rebasing this, so if @desbo and @MatthewJWalls can have a final look 👍

**Comment by regiskuckaertz** [comment]
ping @SiAdcock @desbo @sndrs @MatthewJWalls

**Review by sndrs — APPROVED** [review]
ltgm! 👾

**Pull request merged**